### PR TITLE
Add TestableVSCredentialProvider

### PR DESCRIPTION
**Pull request by pspill**
Test providers for use with NuGet/Home#2909

### CODE REVIEW

File: test/TestExtensions/TestableVSCredentialProvider/TestableVSCredentialProvider.csproj [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <OutputPath>..\..\..\artifacts\TestableVSCredentialProvider\14.0\Debug\</OutputPath>
+    <DefineConstants>TRACE;DEBUG;VS14</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <NoWarn>1762</NoWarn>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <DeployExtension>False</DeployExtension>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(VisualStudioVersion)' == '14.0'">
+    <OutputPath>..\..\..\artifacts\TestableVSCredentialProvider\14.0\Debug\</OutputPath>
```
Comment (alpaix) [inline test/TestExtensions/TestableVSCredentialProvider/TestableVSCredentialProvider.csproj]: Default values are defined in `common.props` file. Please import it here as well.

**Review by alpaix — CHANGES_REQUESTED** [review]
Please update the project file to import default values.

File: test/TestExtensions/TestableVSCredentialProvider/TestableVSCredentialProvider.csproj (hunk 1 above)
Comment (pspill) [reply test/TestExtensions/TestableVSCredentialProvider/TestableVSCredentialProvider.csproj]: Thanks @alpaix, imported and removed a bunch of unnecessary stuff from the .csproj

**Review by alpaix — APPROVED** [review]

**Comment by joelverhagen** [comment]
Is there some steps on how to use with along with the VSIX? Can both be used in the same experimental instance?

**Comment by pspill** [comment]
@joelverhagen added a readme explaining usage

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**